ansible/deploy-pulsar.yaml

```diff
@@ -0,0 +1,166 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+- name: Format and mount disks
+  hosts: pulsar
+  connection: ssh
+  become: true
+  tasks:
+    - shell: >
+        sudo tuned-adm profile latency-performance &&
+        mkfs.xfs /dev/nvme0n1 &&
+        mkfs.xfs /dev/nvme1n1 &&
+        mkdir -p /mnt/journal &&
+        mkdir -p /mnt/storage &&
+        mount -o defaults,noatime,nodiscard,nobarrier /dev/nvme0n1 /mnt/journal &&
+        mount -o defaults,noatime,nodiscard,nobarrier /dev/nvme1n1 /mnt/storage
+      args:
+        creates: /mnt/journal
+
+- name: Pulsar setup
+  hosts: all
+  connection: ssh
+  become: true
+  tasks:
+    - name: Install RPM packages
+      yum: pkg={{ item }} state=latest
+      with_items:
+          - wget
+          - java
+          - sysstat
+          - vim
+
+    - set_fact:
+        zookeeperServers: "{{ groups['zookeeper']|map('extract', hostvars, ['ansible_default_ipv4', 'address'])|map('regex_replace', '(.*)', '\\1:2181') | join(',') }}"
+        serviceUrl: "pulsar://{{ hostvars[groups['pulsar'][0]].private_ip }}:6650/"
```

Comment:
I don't know if terraform supports route53, but it might be more elegant to set this up via an internal domain name.